lib/client/util/__mocks__/Geolocation.js

```diff
@@ -0,0 +1,37 @@
+var path = require.requireActual('../Geolocation');
+
+var geoMock = jest.genMockFromModule('../Geolocation');
+
+var _coords = null;
+var _enabled = false;
+function __setLocation(lat, lon) {
+  _enabled = true;
+  _coords = { latitude: lat, longitude: lon };
+}
+
+function __enable() {
+  _enabled = true;
+}
+
+function __disable() {
+  _coords = null;
+  _enabled = false;
+}
+
+function getCurrentPosition(successCallback, errorCallback) {
+  if(_enabled) {
+    if(_coords) {
+      successCallback({ coords: _coords });
+    }
```

Comment:
It seems that in this specific scenario, both options suck. I'm not really sure if there's an ideal solution. Maybe reaching out to a React-specific slack or irc channel and asking what people do to test async click handlers?

Come to think of it, people may treat `navigator.geolocation.getCurrentLocation` as "data fetching" and do it outside the actual component as an action handler, then the location could be fed back to the component as props. Then it doesn't really encapsulate the location stuff, though. :frowning: